### ISSUE TITLE
Optionally install cla-assistant to use couchbase db instead of mongodb 

### ISSUE THREAD

**Issue by codezork** [cla-assistant/cla-assistant#516]
Because of mongodb's changed license terms, one of our clients wants to use cla-assistant for their repository with a couchbase database. The idea would be to use the node.js ottoman ODM for the database models rather than mongoose via an installation switch. Would a contribution like this be of interest?

**Comment by KharitonOff** [comment]
Yes, sure! 😄 We are always open for valuable contributions. The idea sounds great. We have also got some similar requests but we were not able to implement it sofar. 

Let us know if you need any advice from our side.

**Comment by codezork** [comment]
And here is the first request for an advice 😃 How can I test the callback URL <PROTOCOL>://<HOST>:<PORT>/auth/github/callback against a localhost installation? I used smee.io (https://smee.io/) but the callback path is simply concatenated to my url. So no way with smee. Any suggestions?

**Comment by codezork** [comment]
ahh must escape the greater and less signs, sorry.

**Comment by ibakshay** [comment]
> ahh must escape the greater and less signs, sorry.

😄  So, all good now ? @codezork

**Comment by KharitonOff** [comment]
no problem we use in our case ngrok.io But I can imagine that smee.io is a better solution.

**Comment by codezork** [comment]
as I understood GitHub adds the path for the callback to my given url. can ngrok.io handle this? Smee cannot it seems.

**Comment by codezork** [comment]
..or at least checks that the path is given

**Comment by codezork** [comment]
My mac: „ngrok“ kann nicht geöffnet werden, da es von einem nicht verifizierten Entwickler stammt.

**Comment by KharitonOff** [comment]
hmm, ok. We had no problem with the path of githubs callback. 
Theese are my env varaibles for the setup with ngrok:
```
PORT=5000
PROTOCOL=https
HOST=<mydomain at ngrok>.eu.ngrok.io
```
And in GitHub the "Authorization Callback URL": `https://<mydomain at ngrok>.ngrok.io`

Another alternative could be the serveo.net

**Comment by codezork** [comment]
No luck today: Serveo is temporarily disabled due to phishing. I'll try again on Monday. Thank you!

**Comment by codezork** [comment]
Got ngrok running and tried alike your settings:
                "PORT": "5000",
                "PROTOCOL": "https", 
                "HOST": "https://b65c9cb6.ngrok.io", 
                "MONGODB": "mongodb://cla_assistant:cla_assistant@localhost:27017/cla_assistant", 
                "GITHUB_CLIENT": "ab45ba13e3b04a4dec84", 
                "GITHUB_SECRET": "d413dbc9ae511ab1d84700e8a29443d37edad36f", 
                "GITHUB_USER": "*****", 
                "GITHUB_PASS": "*****" 

In Github:
Homepage URL: https://0d8d054f.ngrok.io
Authorization callback URL: https://0d8d054f.ngrok.io

Github answers when trying to sign in:
https://0d8d054f.ngrok.io/?error=redirect_uri_mismatch&error_description=The%20redirect_uri%20MUST%20match%20the%20registered%20callback%20URL%20for%20this%20application.&error_uri=https:%2F%2Fdeveloper.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch

You wrote in your readme:

Setup your own instance of CLA assistant
Register an application on GitHub. The callback URL needs to be of the form
.../auth/github/callback

I have not done that as passport rejects this callback

Could you imagine what I'm doing wrong?

**Comment by codezork** [comment]
works with grok!

**Comment by ibakshay** [comment]
@codezork  Sounds great 😃

**Comment by codezork** [comment]
It turned out that Couchbase also has a difficult license. So Apache CouchDB will come as alternative. Is this ok for you too?

**Comment by KharitonOff** [comment]
Yes, it is ok from our side

**Comment by codezork** [comment]
Made pull request. CI/CD failed for reasons I do not understand. Could you give a hint?

**Comment by KharitonOff** [comment]
I am sorry for late response. Please, give us some time to look into the PR and check on possible root cause for the failing Build... Probably there are failing unit test. It is important for us to keep a good unit test coverage also for the new features. We'll look into it and let you know.
